wp-parsely-3.0/src/class-parsely.php

```diff
@@ -763,12 +774,17 @@ private function get_top_level_term( int $term_id, string $taxonomy_name ) {
 	 *
 	 * @param int    $post_id The post id you're interested in.
 	 * @param string $taxonomy_name The name of the taxonomy.
-	 * @return string Name of the custom taxonomy.
+	 * @return string|false Name of the custom taxonomy.
 	 */
-	private function get_bottom_level_term( int $post_id, string $taxonomy_name ): string {
-		$terms    = get_the_terms( $post_id, $taxonomy_name );
-		$term_ids = is_array( $terms ) ? wp_list_pluck( $terms, 'term_id' ) : array();
-		$parents  = is_array( $terms ) ? array_filter( wp_list_pluck( $terms, 'parent' ) ) : array();
+	private function get_bottom_level_term( int $post_id, string $taxonomy_name ) {
+		$terms = get_the_terms( $post_id, $taxonomy_name );
+
+		if ( ! is_array( $terms ) ) {
+			return false;
```

Comment:
Could we at least keep the type consistency, and return an empty string? What advantage is there to returning boolean `false`?

Reply:
Actually we want to do this to keep consistency with `get_top_level_term` and the check on line 736.

Reply:
The check on line 736 can check for string length instead, or a `=== ''`.